frontend_lib/src/component/TinyEditor/TinyEditor.jsx

```diff
@@ -0,0 +1,400 @@
+import React, { useEffect, useRef } from 'react'
+import { translate } from 'react-i18next'
+import PropTypes from 'prop-types'
+import i18n from '../../i18n.js'
+import { Editor } from '@tinymce/tinymce-react'
+
+import {
+  getAvatarBaseUrl,
+  handleFetchResult
+} from '../../helper.js'
+import {
+  getMyselfKnownContents
+} from '../../action.async.js'
+import {
+  CUSTOM_EVENT
+} from '../../customEvent.js'
+
+// require('./TinyEditor.styl') // see https://github.com/tracim/tracim/issues/1156
+
+const advancedToolBar = [
+  'formatselect alignleft aligncenter alignright alignjustify | ',
+  'bold italic underline strikethrough | forecolor backcolor | ',
+  'link customInsertImage emoticons charmap | bullist numlist outdent indent | table | ',
+  'code codesample | insert | removeformat | customFullscreen help'
+].join('')
+
+const simpleToolBar = 'bold italic underline | bullist numlist'
+
+const handleFileSelected = (e, editorRef) => {
+  const files = Array.from(e.target.files)
+  base64EncodeAndTinyMceInsert(editorRef, files)
+}
+
+const base64EncodeAndTinyMceInsert = (editorRef, files) => {
+  for (let i = 0; i < files.length; i++) {
+    if (files[i].size > 1000000) {
+      files[i].allowed = globalThis.confirm(
+        i18n.t(
+          '{{filename}} is bigger than 1MB, this may take a while to upload, do you want to continue?',
+          { filename: files[i].name }
+        )
+      )
+    }
+  }
+
+  for (let i = 0; i < files.length; i++) {
+    if (files[i].allowed !== false && files[i].type.match('image.*')) {
+      const img = document.createElement('img')
+      const fr = new globalThis.FileReader()
+
+      fr.readAsDataURL(files[i])
+      fr.onloadend = function (e) {
+        img.src = e.target.result
+        editorRef.current.execCommand('mceInsertContent', false, img.outerHTML)
+      }
+    }
+  }
+}
+
+export const TinyEditor = props => {
+  const editorRef = useRef(null)
+  const inputRef = useRef(null)
+
+  let defaultRoleList = []
+
+  const toolbar = props.isAdvancedEdition ? advancedToolBar : simpleToolBar
+  // NOTE - MP - 2023-01-10 - Changing the key allow reloading the Editor component this allows us
+  // to change the toolbar
+  const editorKey = props.isAdvancedEdition ? 'advanced_key' : 'simple_key'
+
+  useEffect(() => {
```

Comment:
I'm unsure a `useEffect()` is helpful here, what is the goal to do this instead of directly computing `defaultRoleList` in the component? (`const defaultRoleList = props.roleList.map(...)`
If it is a question of optimization, the [useMemo()](https://reactjs.org/docs/hooks-reference.html#usememo) effect seems a better pick.